server/src/main/java/org/elasticsearch/common/lucene/search/MultiPhrasePrefixQuery.java

```diff
@@ -299,6 +302,15 @@ private static boolean termArraysEquals(List<Term[]> termArrays1, List<Term[]> t
     @Override
     public void visit(QueryVisitor visitor) {
         if (visitor.acceptField(field)) {
+            for (int i = 0; i < termArrays.size() - 1; i++) {
+                visitor.consumeTerms(this, termArrays.get(i));
+            }
+            for (Term term : termArrays.get(termArrays.size() - 1)) {
+                visitor.consumeTermsMatching(this, field, () -> {
+                    Automaton a = PrefixQuery.toAutomaton(term.bytes());
+                    return new CompiledAutomaton(a).runAutomaton;
+                });
```

Comment:
I wonder if this automaton will respect `maxExpansions` parameter?  Or since we expect mostly use highlighter based on matches this is not important?